src/telliot_feeds/reporters/reporter360.py

```diff
@@ -0,0 +1,181 @@
+        wait_period: int = 7,
+    ) -> None:
+
+        self.endpoint = endpoint
+        self.oracle = oracle
+        self.token = token
+        self.autopay = autopay
+        self.stake: Optional[int] = stake  # type: ignore
+        self.datafeed = datafeed
+        self.chain_id = chain_id
+        self.acct_addr = to_checksum_address(account.address)
+        self.last_submission_timestamp = 0
+        self.expected_profit = expected_profit
+        self.transaction_type = transaction_type
+        self.gas_limit = gas_limit
+        self.max_fee = max_fee
+        self.wait_period = wait_period
+        self.priority_fee = priority_fee
+        self.legacy_gas_price = legacy_gas_price
+        self.gas_price_speed = gas_price_speed
+        self.autopaytip = 0
+        self.staked_amount: Optional[float] = None
+        self.staker_info: Optional[StakerInfo] = None
+        self.allowed_stake_amount = 0
+
+    async def ensure_staked(self) -> Tuple[bool, ResponseStatus]:
+        # get oracle required stake amount
+        stake, status = await self.oracle.read("getStakedAmount")
+
+        if (not status.ok) or (stake is None):
+            msg = "Unable to read current stake amount"
+            return False, error_status(msg, log=logger.info)
+
+        # get accounts current stake total
+        staker_info, status = await self.oracle.read(
+            "getStakerInfo",
+            _stakerAddress=self.acct_addr,
+        )
+
+        if (not status.ok) or (staker_info is None):
+            msg = "Unable to read reporters staker info"
+            return False, error_status(msg, log=logger.info)
+
+        if self.stake is None:
+            self.stake = stake
+        else:
+            msg = "Stake amount has changed due to TRB price change."
+            logger.info(msg)
+
+        if self.staker_info is None:
+            self.staker_info = StakerInfo(*staker_info)
+        elif self.staker_info.stake_balance > staker_info[1]:
+            # update balance after depositing more stake
+            self.staker_info.stake_balance = staker_info[1]
+            logger.info("your staked balance has decreased and account might be in dispute")
+
+        account_staked_bal = self.staker_info.stake_balance
+
+        if self.stake > account_staked_bal:
+            msg = (
+                "Current stake is low. "
+                # Add more TRB to continue reporting
+                + "Approving & depositing stake."
+            )
+            logger.info(msg)
+
+            gas_price_gwei = await self.fetch_gas_price()
+            if gas_price_gwei is None:
+                return False, error_status("Unable to fetch matic gas price for staking", log=logger.info)
+
+            # diff in required stake and account staked balance
+            stake_diff = self.stake - account_staked_bal
+
+            # check TRB wallet balance!
+            wallet_balance = self.token.read("balanceOf", account=self.acct_addr)
```

Comment:
I think you need error handling for no balance retrieved from this contract read.